megabus/src/main/java/com/bazaarvoice/megabus/tableevents/TableEventProcessor.java

```diff
@@ -0,0 +1,102 @@
+package com.bazaarvoice.megabus.tableevents;
+
+import com.bazaarvoice.emodb.common.uuid.TimeUUIDs;
+import com.bazaarvoice.emodb.kafka.Topic;
+import com.bazaarvoice.emodb.sor.api.Coordinate;
+import com.bazaarvoice.emodb.table.db.eventregistry.TableEventTools;
+import com.bazaarvoice.emodb.table.db.eventregistry.TableEvent;
+import com.bazaarvoice.emodb.table.db.eventregistry.TableEventRegistry;
+import com.bazaarvoice.megabus.MegabusRef;
+import com.codahale.metrics.MetricRegistry;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Iterators;
+import com.google.common.util.concurrent.AbstractScheduledService;
+import com.google.common.util.concurrent.Futures;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+public class TableEventProcessor extends AbstractScheduledService {
```

Comment:
Add class comment - this is the leader elected process to send the READY table events to Megabus Ref topic, and then updating the events as complete in the system table.

Reply:
👍